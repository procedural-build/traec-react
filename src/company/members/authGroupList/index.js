import React from "react";
import { connect } from 'react-redux';
import Traec from 'traec'

import { BSCard, BSBtn } from 'AppSrc/utils/bootstrap';
import CompanyAuthGroupForm from './form'
import AuthGroupItem from './item'


class AuthGroupList extends React.Component { 

    constructor(props) {
        super(props)

        const companyId = props.companyId
        this.fetch = new Traec.Fetch('company_authgroup', 'post', {companyId})
        let {fetchParams, stateParams} = this.fetch.params

        this.state = {
            showForm: false,
            formParams: {
                fetchParams, 
                stateParams,
                initFields: {}
            }
        }

        // Data required from the API for this Component
        this.requiredFetches = [
            new Traec.Fetch('company_authgroup', 'list')
        ]

        // Bindings for actions
        this.onClick = this.onClick.bind(this);
        this.showFormHandler = this.showFormHandler.bind(this)
    }

    componentDidMount() {
        Traec.fetchRequired.bind(this)()
    }

    componentDidUpdate() {
        Traec.fetchRequired.bind(this)()
    }

    /* ACTIONS */

    onClick(e) {
        e.preventDefault();
        this.fetch.toggleForm()
    }

    showFormHandler(e, showForm) {
        this.setState({showForm})
    }

    /* RENDERING */

    render_form() {
        if (!this.state.showForm) { return null }
        let {companyId, dispatch} = this.props
        return (
            <CompanyAuthGroupForm 
                item={null}
                companyId={companyId}
                showFormHandler={this.showFormHandler}
                fetchHandler={new Traec.Fetch('company_authgroup', 'post', {companyId})}
                dispatch={dispatch}
            />
        )
    }

    render(){
        let {companyId, dispatch, authgroups} = this.props
        if (!authgroups) { return null }

        let itemList = authgroups.toList()
            .sortBy( (obj, i) => obj.get('name') )
            .map((item, i) => (
                <AuthGroupItem 
                    key={i} index={i} 
                    item={item} companyId={companyId}
                    dispatch={dispatch} 
                />
            ))
        //
        return (
            <React.Fragment>
            <div className="row">
                <BSCard 
                    widthOffset="col-sm-12"
                    title="Authority Groups"
                    button={<BSBtn onClick={e => this.showFormHandler(e, !this.state.showForm)} text="Add an Authority Group" />}
                    body={itemList}
                    form={this.render_form()}
                />
            </div>
            </React.Fragment>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    let {companyId} = ownProps
    let company = state.getInPath(`entities.companys.byId.${companyId}`)
    let authgroups = state.getInPath(`entities.companyObjects.byId.${companyId}.authgroups`)
    return ({
        company,
        authgroups,
        isAuthenticated: state.getIn('auth.isAuthorized'.split('.'))
    })
}



const mapDispatchToProps = dispatch => {
    return {
        dispatch: dispatch,
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(AuthGroupList);