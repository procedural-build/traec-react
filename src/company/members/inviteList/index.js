import React from "react";
import ReactDOM from "react-dom";
import Traec from 'traec'

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { BSCard } from 'traec-react/utils/bootstrap';
import InviteItem from './item'


class InviteList extends React.Component { 

    constructor(props) {
        super(props)

        this.state = {}

        // Data required from the API for this Component
        this.requiredFetches = [
            new Traec.Fetch('company_invite', 'list')
        ]
    }

    componentDidMount() {
        Traec.fetchRequired.bind(this)()
    }

    componentDidUpdate() {
        Traec.fetchRequired.bind(this)()
    }

    render(){
        let {invites, companyId, dispatch} = this.props
        if (!invites || !invites.size) { return null }

        // Get the list of Invite components
        let itemList = invites.toList()
            .sortBy( (obj, i) => obj.get('created') )
            .map((invite, i) => (
                <InviteItem 
                    key={i} 
                    index={i} 
                    item={invite} 
                    companyId={companyId}
                    dispatch={dispatch}
                />
            ))
        
        return (
            <div className="row">
                <BSCard 
                    widthOffset="col-sm-12"
                    title="Outstanding Invitations"
                    button={null}
                    body={itemList}
                    form={null}
                />
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    let {companyId} = ownProps
    let company = state.getInPath(`entities.companies.byId.${companyId}`)
    let invites = state.getInPath(`entities.companyObjects.byId.${companyId}.invites`)

    return ({
        company,
        invites,
        isAuthenticated: state.getInPath('auth.isAuthorized')
    })
}



const mapDispatchToProps = dispatch => {
    return {
        dispatch: dispatch,
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(InviteList);