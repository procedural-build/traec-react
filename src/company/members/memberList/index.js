import Im from 'traec/immutable';
import React from "react";
import ReactDOM from "react-dom";
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Traec from 'traec'

import { BSCard, BSBtn, BSBtnDropdown } from 'traec-react/utils/bootstrap';

import { objToList } from 'traec-react/utils'
import InviteForm, {inviteFields} from './form';
import {companyPermissionRender} from 'traec/utils/permissions/company'
import MemberItem from './item'


class MemberList extends React.Component { 

    constructor(props) {
        super(props)

        const companyId = props.companyId
        this.fetch = new Traec.Fetch('company_invite', 'post', {companyId})
        let {fetchParams, stateParams} = this.fetch.params

        this.state = {
            formParams: {
                fetchParams, 
                stateParams,
                initFields: {}
            }
        }

        // Data required from the API for this Component
        this.requiredFetches = [
            new Traec.Fetch('company_member', 'list')
        ]

        // Bindings for actions
        this.onClick = this.onClick.bind(this);
    }

    componentDidMount() {
        Traec.fetchRequired.bind(this)()
    }

    componentDidUpdate() {
        Traec.fetchRequired.bind(this)()
    }

    /* ACTIONS */

    onClick(e) {
        e.preventDefault();
        this.fetch.toggleForm()
    }

    /* RENDERING */

    render(){
        let {companyId, members, dispatch} = this.props
        let itemList = objToList(members)
            .sortBy(i => i.getIn(['user', 'first_name']))
            .map((member, i) => (
                <MemberItem key={i} index={i} dispatch={dispatch} member={member} companyId={companyId}/>
            ));

        return (
            <div className="row">
                <BSCard 
                    widthOffset="col-sm-12"
                    title="Members"
                    button={companyPermissionRender(
                        companyId, true, [],
                        (<BSBtn onClick={this.onClick} text="Send Invite" />)
                    )}
                    body={itemList}
                    form={<InviteForm 
                        companyId={companyId}
                        stateParams={this.state.formParams.stateParams} 
                        fetchParams={this.state.formParams.fetchParams}
                        fields={inviteFields}
                    />}
                />
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    let {companyId} = ownProps
    let company = state.getInPath(`entities.companies.byId.${companyId}`)
    const members = state.getInPath(`entities.companyObjects.byId.${companyId}.members`)
    return ({
        company,
        members,
        isAuthenticated: state.getInPath('auth.isAuthorized')
    })
}


const mapDispatchToProps = dispatch => {
    return {
        dispatch: dispatch,
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(MemberList);