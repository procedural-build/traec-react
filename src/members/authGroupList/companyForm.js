import React from "react";
import ReactDOM from "react-dom";

export const authGroupFields = {
  name: { value: "", class: "col col-sm-11 mb-1", label: "", placeholder: "Name" },
  is_admin: { value: false, class: "col col-sm-1 mb-1", label: "Administrator", inputType: "checkbox" }
};

export default class CompanyAuthGroupForm extends React.Component {
  constructor(props) {
    super(props);

    this.operations = ["CREATE", "READ", "UPDATE", "DELETE"];
    this.viewObjectTypes = {
      COMPANY_COMPANY: "Business Units",
      COMPANY_MEMBER: "Corporate Users",
      COMPANY_PROJECT: "Corporate Projects",
      COMPANY: "Corporate Details",
      COMPANY_REPORT: "Corporate Dashboard"
    };

    let { item } = props;
    let initActionList = item ? item.getIn(["policy_json", "actions"]) || [] : [];
    this.state = {
      uid: item ? item.get("uid") : null,
      name: item ? item.get("name") : "",
      is_admin: item ? item.get("is_admin") : false,
      actions: initActionList.reduce((obj, i) => {
        obj[i] = true;
        return obj;
      }, {})
    };

    this.changeAction = this.changeAction.bind(this);
    this.toggleAdmin = this.toggleAdmin.bind(this);
    this.changeTextInput = this.changeTextInput.bind(this);
    this.submitData = this.submitData.bind(this);
  }

  changeAction(e, action) {
    let actions = Object.assign({}, this.state.actions);
    actions[action] = !actions[action];
    this.setState({ actions });
  }

  toggleAdmin(e) {
    this.setState({ is_admin: !this.state.is_admin });
  }

  changeTextInput(e) {
    e.preventDefault();
    this.setState({ [e.target.name]: e.target.value });
  }

  submitData(e) {
    e.preventDefault();
    let { companyId, fetchHandler } = this.props;
    // Construct the auth object that will be sent
    let actions = [];
    if (!this.state.is_admin) {
      for (let [action, isAction] of Object.entries(this.state.actions)) {
        if (isAction) {
          actions.push(action);
        }
      }
    }
    let data = {
      name: this.state.name,
      is_admin: this.state.is_admin,
      policy_json: { actions }
    };
    // The fetchHandler to use should be passed in as a prop
    fetchHandler.update({ companyId, authGroupId: this.state.uid });
    fetchHandler.updateFetchParams({ body: data });
    fetchHandler.dispatch();
  }

  render_actions_table_rows() {
    let rows = [];
    let counter = 0;
    for (let [objKey, name] of Object.entries(this.viewObjectTypes)) {
      let colCells = this.operations.map((operation, i) => {
        let action = `${operation}_${objKey}`;
        let isChecked = this.state.actions[action] || false;
        return (
          <td key={counter++} className="text-center" style={{ backgroundColor: isChecked ? "#99EB99" : "#EB9999" }}>
            <input type="checkbox" checked={isChecked} onChange={e => this.changeAction(e, action)} />
          </td>
        );
      });
      // Add the row header
      colCells.unshift(
        <th key={counter++} scope="row">
          {name}
        </th>
      );
      // Add this to the rows
      rows.push(<tr key={counter++}>{colCells}</tr>);
    }
    return rows;
  }

  render_actions_table() {
    if (this.state.is_admin) {
      return null;
    }
    let colHeadings = this.operations.map((o, i) => (
      <th key={i} scope="col" width="15%" className="text-center">
        {o}
      </th>
    ));
    colHeadings.unshift(<th key={colHeadings.length} scope="col" />);
    return (
      <div className="form-group row mb-0">
        <div className="col-sm-12">
          <table className="table table-sm">
            <thead>
              <tr>{colHeadings}</tr>
            </thead>
            <tbody>{this.render_actions_table_rows()}</tbody>
          </table>
        </div>
      </div>
    );
  }

  render() {
    const i = this.props.index;
    const item = this.props.item;

    return (
      <React.Fragment>
        <form>
          <div className="form-group row mb-0">
            <div className="col-sm-11">
              <input
                type="text"
                name="name"
                className="form-control-plaintext"
                value={this.state.name}
                onChange={this.changeTextInput}
                placeholder="Enter a name for this Role"
              />
            </div>
            <div className="form-check col-sm-1">
              <input
                className="form-check-input"
                type="checkbox"
                id="adminCheck"
                checked={this.state.is_admin}
                onChange={this.toggleAdmin}
              />
              <label className="form-check-label" htmlFor="adminCheck">
                Admin?
              </label>
            </div>
          </div>
          {this.render_actions_table()}
          <button className="btn btn-sm btn-default" onClick={e => this.props.showFormHandler(e, false)}>
            Close
          </button>
          <button className="btn btn-sm btn-primary float-right" onClick={this.submitData}>
            Save
          </button>
          <div style={{ clear: "both" }} />
        </form>
        <hr />
      </React.Fragment>
    );
  }
}
