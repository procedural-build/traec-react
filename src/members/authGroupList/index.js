import React from "react";
import { connect } from "react-redux";
import Traec from "traec";
import { BSCard, BSBtn } from "traec-react/utils/bootstrap";
import { ProjectAuthGroupForm } from "./projectForm";
import CompanyAuthGroupForm from "./companyForm";
import AuthGroupItem from "./item";

export class AuthGroupList extends React.Component {
  constructor(props) {
    super(props);

    if (props.projectId) {
      const projectId = props.projectId;
      this.fetch = new Traec.Fetch("project_authgroup", "post", { projectId });
      this.requiredFetches = [new Traec.Fetch("project_authgroup", "list")];
    } else if (props.companyId) {
      const companyId = props.companyId;
      this.fetch = new Traec.Fetch("company_authgroup", "post", { companyId });
      this.requiredFetches = [new Traec.Fetch("company_authgroup", "list")];
    }

    let { fetchParams, stateParams } = this.fetch.params;

    this.state = {
      showForm: false,
      formParams: {
        fetchParams,
        stateParams,
        initFields: {}
      }
    };

    this.onClick = this.onClick.bind(this);
    this.showFormHandler = this.showFormHandler.bind(this);
  }

  componentDidMount() {
    Traec.fetchRequired.bind(this)();
  }

  componentDidUpdate() {
    Traec.fetchRequired.bind(this)();
  }

  onClick(e) {
    e.preventDefault();
    this.fetch.toggleForm();
  }

  showFormHandler(e, showForm) {
    this.setState({ showForm });
  }

  /* RENDERING */

  renderForm() {
    if (!this.state.showForm) {
      return null;
    }

    if (this.props.projectId) {
      return (
        <ProjectAuthGroupForm
          item={null}
          projectId={this.props.projectId}
          showFormHandler={this.showFormHandler}
          fetchHandler={this.fetch}
          dispatch={this.props.dispatch}
        />
      );
    }

    if (this.props.companyId) {
      return (
        <CompanyAuthGroupForm
          item={null}
          companyId={this.props.companyId}
          showFormHandler={this.showFormHandler}
          fetchHandler={this.fetch}
          dispatch={this.props.dispatch}
        />
      );
    }
  }

  render() {
    let { authGroups, projectId, companyId, dispatch } = this.props;
    if (!authGroups) {
      return null;
    }

    let itemList = authGroups
      .toList()
      .sortBy((obj, i) => obj.get("name"))
      .map((item, i) => (
        <AuthGroupItem key={i} index={i} item={item} projectId={projectId} companyId={companyId} dispatch={dispatch} />
      ));

    return (
      <div className="row">
        <BSCard
          widthOffset="col-sm-12"
          title="Authority Groups"
          button={<BSBtn onClick={e => this.showFormHandler(e, !this.state.showForm)} text="Add an Authority Group" />}
          body={itemList}
          form={this.renderForm()}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  let { projectId, companyId } = ownProps;

  let project = null;
  let company = null;
  let authGroups = null;

  if (projectId) {
    project = state.getInPath(`entities.projects.byId.${projectId}`);
    authGroups = state.getInPath(`entities.projectObjects.byId.${projectId}.authGroups`);
  }

  if (companyId) {
    company = state.getInPath(`entities.companies.byId.${companyId}`);
    authGroups = state.getInPath(`entities.companyObjects.byId.${companyId}.authGroups`);
  }
  let isAuthenticated = state.getInPath("auth.isAuthenticated");
  return { project, company, authGroups, isAuthenticated };
};

const mapDispatchToProps = dispatch => {
  return {
    dispatch: dispatch
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthGroupList);
