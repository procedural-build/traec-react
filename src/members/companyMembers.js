import React from "react";
import { connect } from "react-redux";
import { companyPermissionRender } from "traec/utils/permissions/company";

import MemberList from "./memberList";
import InviteList from "./inviteList";
import AuthGroupList from "./authGroupList";

export class CompanyMembers extends React.Component {
  render() {
    const { companyId, company } = this.props;
    if (!company) {
      return null;
    }
    return (
      <React.Fragment>
        <h3>Company Members</h3>
        <p>{company.get("name")}</p>

        {/*Render the members panel if allowed */}
        {companyPermissionRender(companyId, false, ["READ_COMPANY_MEMBER"], <MemberList companyId={companyId} />)}

        {/*Render the invites panel if allowed */}
        {companyPermissionRender(companyId, true, [], <InviteList companyId={companyId} />)}

        {/*Render the authGroup panel if allowed */}
        {companyPermissionRender(companyId, true, [], <AuthGroupList companyId={companyId} />)}
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const { companyId } = ownProps.match.params;
  let company = state.getInPath(`entities.companies.byId.${companyId}`);
  return { companyId, company };
};

const mapDispatchToProps = dispatch => {
  return {
    dispatch: dispatch
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CompanyMembers);
