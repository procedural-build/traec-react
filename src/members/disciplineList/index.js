import React from "react";
import { connect } from "react-redux";
import Traec from "traec";
import { BSCard, BSBtn } from "traec-react/utils/bootstrap";
import DisciplineForm, { disciplineFields } from "./form";
import { ProjectPermission } from "traec/utils/permissions/project";
import DisciplineItem from "./item";

export class DisciplineList extends React.Component {
  constructor(props) {
    super(props);

    const projectId = props.projectId;
    this.fetch = new Traec.Fetch("project_discipline", "post", { projectId });
    let { fetchParams, stateParams } = this.fetch.params;

    this.state = {
      formParams: {
        fetchParams,
        stateParams,
        initFields: {}
      }
    };

    this.requiredFetches = [new Traec.Fetch("project_discipline", "list")];

    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    e.preventDefault();
    this.fetch.toggleForm();
  }

  componentDidMount() {
    Traec.fetchRequired.bind(this)();
  }

  render() {
    let {
      projectId,
      dispatch,
      disciplines,
      tree,
      title = "Project Suppliers",
      buttonText = "Add a Supplier"
    } = this.props;
    if (!disciplines) {
      return null;
    }

    let rootChildren = [...tree["root"].children];
    let rootItems = Traec.Im.fromJS(rootChildren.map(id => tree[id].obj));

    let itemList = rootItems
      .sortBy(i => i.get("name"))
      .filter(i => !i.get("approver"))
      .map((discipline, i) => (
        <DisciplineItem key={i} index={i} item={discipline} tree={tree} projectId={projectId} dispatch={dispatch} />
      ));

    let fields = {
      name: Object.assign({}, disciplineFields.name),
      auth: Object.assign({}, disciplineFields.auth),
      approver: Object.assign({}, disciplineFields.approver)
    };

    return (
      <div className="row">
        <BSCard
          widthOffset="col-sm-12"
          title={title}
          button={
            <ProjectPermission projectId={projectId} requiresAdmin={true}>
              <BSBtn onClick={this.onClick} text={buttonText} />
            </ProjectPermission>
          }
          body={itemList}
          form={
            <DisciplineForm
              projectId={this.props.projectId}
              stateParams={this.state.formParams.stateParams}
              fetchParams={this.state.formParams.fetchParams}
              fields={fields}
            />
          }
        />
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  let { projectId } = ownProps;
  let project = state.getInPath(`entities.projects.byId.${projectId}`);
  let disciplines = state.getInPath(`entities.projectObjects.byId.${projectId}.disciplines`);
  // Make a tree of the discipline approval heirarchy
  let tree = {};
  if (disciplines) {
    disciplines = disciplines.filter(discipline => !discipline.getInPath("meta_json.user_discipline"));
    for (let [itemId, item] of disciplines) {
      let approverId = item.get("approver") || "root";
      // Add the item to the tree
      let treeItem = tree[itemId] || { obj: item, children: new Set() };
      Object.assign(treeItem, { obj: item });
      Object.assign(tree, { [itemId]: treeItem });
      // Add this to the children set of the parent
      treeItem = tree[approverId] || { obj: null, children: new Set() };
      treeItem.children.add(itemId);
      Object.assign(tree, { [approverId]: treeItem });
    }
  }
  // Return this
  return { project, disciplines, tree };
};

const mapDispatchToProps = dispatch => {
  return {
    dispatch: dispatch
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(DisciplineList);
