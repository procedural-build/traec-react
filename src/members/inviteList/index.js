import React from "react";
import Traec from "traec";
import { connect } from "react-redux";
import { BSCard } from "traec-react/utils/bootstrap";
import InviteItem from "./item";

export class InviteList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};

    // Data required from the API for this Component
    if (props.projectId) {
      this.requiredFetches = [new Traec.Fetch("project_invite", "list")];
    } else if (props.companyId) {
      this.requiredFetches = [new Traec.Fetch("company_invite", "list")];
    }
  }

  componentDidMount() {
    Traec.fetchRequired.bind(this)();
  }

  componentDidUpdate() {
    Traec.fetchRequired.bind(this)();
  }

  render() {
    let { invites, companyId, projectId, dispatch } = this.props;
    if (!invites || !invites.size) {
      return null;
    }

    // Get the list of Invite components
    let itemList = invites
      .toList()
      .sortBy((obj, i) => obj.get("created"))
      .map((invite, i) => (
        <InviteItem key={i} index={i} item={invite} companyId={companyId} projectId={projectId} dispatch={dispatch} />
      ));

    return (
      <div className="row">
        <BSCard widthOffset="col-sm-12" title="Outstanding Invitations" button={null} body={itemList} form={null} />
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  let { companyId, projectId } = ownProps;
  let project = null;
  let company = null;
  let invites = null;

  if (ownProps.projectId) {
    project = state.getInPath(`entities.projects.byId.${projectId}`);
    invites = state.getInPath(`entities.projectObjects.byId.${projectId}.invites`);
    let disciplines = state.getInPath(`entities.projectObjects.byId.${projectId}.disciplines`);

    if (invites && disciplines) {
      for (let [key, invite] of invites) {
        let disciplineId = invite.get("project_discipline");
        invites = invites.setInPath(`${key}.project_discipline`, disciplines.get(disciplineId));
      }
    }
  } else if (ownProps.companyId) {
    company = state.getInPath(`entities.companies.byId.${companyId}`);
    invites = state.getInPath(`entities.companyObjects.byId.${companyId}.invites`);
  }

  return {
    project,
    company,
    invites,
    isAuthenticated: state.getInPath("auth.isAuthorized")
  };
};

const mapDispatchToProps = dispatch => {
  return {
    dispatch: dispatch
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(InviteList);
