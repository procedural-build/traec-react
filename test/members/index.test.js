import React from "react";
import { shallow } from "enzyme";
import { ProjectMembers } from "../../src/members/projectMembers";
import { CompanyMembers } from "../../src/members/companyMembers";
import { MemberList } from "../../src/members/memberList";
import { InviteList } from "../../src/members/inviteList";
import { DisciplineList } from "../../src/members/disciplineList";
import { CompanyAuthGroupForm } from "../../src/members/authGroupList/companyForm";
import { ProjectAuthGroupForm } from "../../src/members/authGroupList/projectForm";
import { AuthGroupList } from "../../src/members/authGroupList";
import { Link } from "react-router-dom";
import Octicon from "react-octicon";
import Im from "traec/immutable";

import TestRenderer from "react-test-renderer";
jest.mock("react-router-dom", () => ({ Link: "Link" }));

describe("ProjectMembers", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<ProjectMembers />)));

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });

  it("should render a <h3 />", () => {
    expect(wrapper.find("h3").length).toEqual(0);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(<ProjectMembers />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("CompanyMembers", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<CompanyMembers />)));

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });

  it("should render a <h3 />", () => {
    expect(wrapper.find("h3").length).toEqual(0);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(<CompanyMembers />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("MemberList", () => {
  let wrapper;

  const dispatch = jest.fn();
  const member = {};
  const companyId = jest.fn();
  const projectId = jest.fn();
  beforeEach(
    () =>
      (wrapper = shallow(
        <MemberList key={0} index={0} dispatch={dispatch} member={member} companyId={companyId} projectId={projectId} />
      ))
  );

  it("should contain MemberList", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(
      <MemberList key={0} index={0} dispatch={dispatch} member={member} companyId={companyId} projectId={projectId} />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("InviteList", () => {
  let wrapper;
  const dispatch = jest.fn();
  const invites = {};
  const companyId = jest.fn();
  const projectId = jest.fn();

  beforeEach(
    () =>
      (wrapper = shallow(
        <InviteList dispatch={dispatch} invites={invites} companyId={companyId} projectId={projectId} />
      ))
  );

  it("should contain InviteList", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(
      <InviteList dispatch={dispatch} invites={invites} companyId={companyId} projectId={projectId} />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("DisciplineList", () => {
  let wrapper;
  const key = jest.fn();
  const index = jest.fn();
  const projectId = jest.fn();
  const discipline = jest.fn();
  const tree = jest.fn();
  const dispatch = jest.fn();

  beforeEach(
    () =>
      (wrapper = shallow(
        <DisciplineList
          key={key}
          index={index}
          item={discipline}
          tree={tree}
          projectId={projectId}
          dispatch={dispatch}
        />
      ))
  );
  it("should contain DisciplineList", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(
      <DisciplineList key={key} index={index} item={discipline} tree={tree} projectId={projectId} dispatch={dispatch} />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("ProjectAuthGroupForm", () => {
  let wrapper;
  const item = jest.fn();
  const projectId = jest.fn();
  const showFormHandler = jest.fn();
  const fetchHandler = jest.fn();
  const dispatch = jest.fn();

  beforeEach(
    () =>
      (wrapper = shallow(
        <AuthGroupList
          item={item}
          projectId={projectId}
          showFormHandler={showFormHandler}
          fetchHandler={fetchHandler}
          dispatch={dispatch}
        />
      ))
  );

  it("should contain ProjectAuthGroupForm", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(
      <AuthGroupList
        item={item}
        projectId={projectId}
        showFormHandler={showFormHandler}
        fetchHandler={fetchHandler}
        dispatch={dispatch}
      />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("CompanyAuthGroupForm", () => {
  let wrapper;
  const item = jest.fn();
  const companyId = jest.fn();
  const showFormHandler = jest.fn();
  const fetchHandler = jest.fn();
  const dispatch = jest.fn();

  beforeEach(
    () =>
      (wrapper = shallow(
        <AuthGroupList
          item={item}
          companyId={companyId}
          showFormHandler={showFormHandler}
          fetchHandler={fetchHandler}
          dispatch={dispatch}
        />
      ))
  );

  it("should contain CompanyAuthGroupForm", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(
      <AuthGroupList
        item={item}
        companyId={companyId}
        showFormHandler={showFormHandler}
        fetchHandler={fetchHandler}
        dispatch={dispatch}
      />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe("AuthGroupList", () => {
  let wrapper;
  const key = jest.fn();
  const index = jest.fn();
  const item = jest.fn();
  const projectId = jest.fn();
  const dispatch = jest.fn();

  beforeEach(
    () =>
      (wrapper = shallow(
        <AuthGroupList key={key} index={index} item={item} projectId={projectId} dispatch={dispatch} />
      ))
  );

  it("should contain AuthGroupList", () => {
    expect(wrapper.find("div").length).toEqual(0);
  });

  it("should match snapshot", () => {
    const testRenderer = TestRenderer.create(
      <AuthGroupList key={key} index={index} item={item} projectId={projectId} dispatch={dispatch} />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
