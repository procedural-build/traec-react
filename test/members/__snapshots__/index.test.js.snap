// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthGroupList should match snapshot 1`] = `null`;

exports[`CompanyAuthGroupForm should match snapshot 1`] = `null`;

exports[`CompanyMembers should match snapshot 1`] = `null`;

exports[`DisciplineList should match snapshot 1`] = `null`;

exports[`DisciplineList should match snapshot 2`] = `null`;

exports[`InviteList should match snapshot 1`] = `null`;

exports[`MemberList should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
    id=""
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <h5
          className="card-title float-left"
        >
          Members
        </h5>
      </div>
      <div
        className="card-body"
      >
        
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectAuthGroupForm should match snapshot 1`] = `null`;

exports[`ProjectMembers should match snapshot 1`] = `""`;
